File: robots/ruby/lib/machine.rb
class Machine
  attr_reader :name, :bin, :location

  def initialize(name, location)
    @name = name
    @location = location
  end

  def take
    result = @bin
    @bin = nil
    return result
  end	

  def put(bin)
    @bin = bin
  end
end

File: robots/javascript/machine.js
function Machine(name, location) {
	this.name = name;
	this.bin = null;
	this.location = location;
}

Machine.prototype = {
  take: function() {
    var result = this.bin
    this.bin = null;
    return result;
  },

  put: function(bin) {
    this.bin = bin;
  },

  toString: function() {
  	return '[object Machine]';
  }
}
